### ISSUE TITLE
Badger Firefox fails on read.cash

### ISSUE THREAD

**Issue by readcash** [Bitcoin-com/badger#215]
**Describe the bug**
Trying to log in with Badger Wallet Firefox results in this:

![image](https://user-images.githubusercontent.com/58006668/69672949-3ea97880-10aa-11ea-8f28-fcc305c2afce.png)

**To Reproduce**
1. Use Badger on Firefox (Mac or reported by user on Ubuntu too)
2. Go to http://read.cash/register
unlock Badger
3. Click "Badger Wallet"
4. Click sign
... no error is shown, but console shows error.

**Expected behavior**
1) It should show an error to user
or
2) It should log in (Badger Chrome works fine), on our side we see only OPTIONS request that results in 200 response. Same server works fine on Badger Chrome.

**Screenshots**
![image](https://user-images.githubusercontent.com/58006668/69672949-3ea97880-10aa-11ea-8f28-fcc305c2afce.png)

**Comment by SpendBCH** [comment]
Hi,

This might be a difference between how Chrome and Firefox handle certain requests, can you you try again in Firefox with an `https` address?

Thanks!

@gitcash 1000 bits for @readcash

**Comment by GitCash** [comment]
Hey **readcash**, user **SpendBCH** tipped you **1000 bits** in Bitcoin Cash ( ~ **$0.212** ).

*[Click here to claim it!](https://gitcash.io/claim/readcash)*

You can also add the "thumbs down" reaction to **SpendBCH**'s comment above to prevent future tips.

**Comment by readcash** [comment]
So far, it's the same!

Badger shows Version 0.7.9 - is this the right version to try or do I need to update it?

Thanks for the tip! :) It'll go towards supporting the authors at https://read.cash

**Comment by readcash** [comment]
> Thanks for the tip! :) It'll go towards supporting the authors at https://read.cash

...or maybe not (see the issues above) 🤷‍♂

**Comment by readcash** [comment]
You can probably try it yourself - it's like 4 clicks: https://read.cash/register -> Understood -> Badger -> Sign

**Comment by SpendBCH** [comment]
Thanks for trying that update, looking into this more it looks like a CORS issue.

Could you please try enabling CORS on your server -- the issue is the request does not come from your own domain and Firefox will reject it.

Hopefully deposits will start going through again soon, gitcash is a great service.

**Comment by readcash** [comment]
Yep, CORS fixes it. Thanks! Can I ask you to add try/catch and an error to badger?... it's really confusing, seems like "nothing is happening" and I'm sure as Badger gets more popular - similar issues will popup (CORS without any error). Thanks!
